src/app/composer/qgscomposermanager.cpp

```diff
@@ -110,14 +115,61 @@ void QgsComposerManager::refreshComposers()
   mComposerListWidget->sortItems();
 
   // Restore selection
-  if ( !selName.isEmpty() )
+  if ( !selectedComposers.isEmpty() )
   {
-    QList<QListWidgetItem*> items = mComposerListWidget->findItems( selName, Qt::MatchExactly );
-    if ( !items.isEmpty() )
+    Q_FOREACH ( QgsComposer* c, selectedComposers )
     {
-      mComposerListWidget->setCurrentItem( items.first() );
+      QMap<QListWidgetItem*, QgsComposer*>::const_iterator i = mItemComposerMap.constBegin();
+      while ( i != mItemComposerMap.constEnd() )
+      {
+	// This composer was selected: reselect it !
+	if ( c == i.value() )
+	{
+	  int inDex = mComposerListWidget->row( i.key() );
+	  QModelIndex selectLine = mComposerListWidget->model()->index( inDex, 0, QModelIndex() );
+	  mComposerListWidget->selectionModel()->select( selectLine, QItemSelectionModel::Select );
+	}
+	++i;
+      }
     }
   }
+  // Select the first item by default
+  else if ( mComposerListWidget->count() > 0 )
```

Comment:
it's also possible that all selected composers were removed. So a better approach might be having a bool `selectionRestored` which is set to true if any composer was reselected from the selectedComposers.

Then change

    else if ( mComposerListWidget->count() > 0 )

to

    if ( !selectionRestored && !mComposerListWidget.isEmpty() )

This would mean deleting all composers would also select the first remaining item.